XRTK-Core/Packages/com.xrtk.core/Services/MixedRealityToolkit.cs

```diff
@@ -1747,14 +1847,15 @@ private static bool CheckServiceMatch(Type interfaceType, string serviceName, Ty
         {
             bool isValid = string.IsNullOrEmpty(serviceName) || serviceInstance.Name == serviceName;
 
-            if ((registeredInterfaceType.Name == interfaceType.Name || serviceInstance.GetType().Name == interfaceType.Name) && isValid)
+            if ((registeredInterfaceType.Name == interfaceType.Name ||
+                 serviceInstance.GetType().Name == interfaceType.Name) && isValid)
             {
                 return true;
             }
 
             var interfaces = serviceInstance.GetType().GetInterfaces();
```

Comment:
I'm wondering if we can replace everything below this with

`interfaceType.IsInstanceOfType(serviceInstance)`

Reply:
Are you planing on implementing / testing this?

Reply:
It was added to the unit tests already

Reply:
I'm not really sure how this will change the behavior. I think I'll just keep it in for now.

Reply:
Actually it was so much better to use my suggestion!  Take a look at the latest changes and let me know what you think.